vector/src/main/java/fr/gouv/tchap/features/home/contact/list/TchapContactListViewModel.kt

```diff
@@ -138,31 +141,29 @@ class TchapContactListViewModel @AssistedInject constructor(@Assisted initialSta
             }
 
             val unresolvedThreePids = mutableListOf<String>()
-            val users = mutableMapOf<String, User>()
-
-            data.associateByTo(users,
-                    { it.matrixId },
-                    {
-                        session.getUser(it.matrixId) ?: run {
-                            unresolvedThreePids.add(it.matrixId)
-                            //Create a temporary user
-                            User(it.matrixId, TchapUtils.computeDisplayNameFromUserId(it.matrixId), null)
-                        }
-                    })
+            val users: MutableMap<String, User> = data.map {
+                var user = session.getUser(it.matrixId)
+                user = if (user == null) {
+                    unresolvedThreePids.add(it.matrixId)
+                    User(it.matrixId, TchapUtils.computeDisplayNameFromUserId(it.matrixId), null)
+                } else {
+                    user
+                }
+                it.matrixId to user
+            }.toMap().toMutableMap()
 
-            setState {
-                copy(localUsers = users.values.toList())
-            }
+            localUsers = users.values.toList()
 
             updateFilteredContacts()
 
             unresolvedThreePids.mapNotNull {
                 tryOrNull { session.resolveUser(it) }
             }.forEach { user -> users[user.userId] = user }
 
+            localUsers = users.values.toList()
+
             setState {
```

Comment:
Don't you think you should call `updateFilteredContacts()` before update the state?